packages/backend-data/src/convert_schema.ts

```diff
@@ -16,22 +29,51 @@ export const isModelSchema = (
   return (
     schema !== null &&
     typeof schema === 'object' &&
+    'transform' in schema &&
     typeof schema.transform === 'function'
   );
 };
 
+/**
+ * Determine if the input schema is a collection of model schemas, and perform type narrowing.
+ * @param schema the schema that might be a collection of model schemas
+ * @returns a boolean indicating whether the schema is a collection of derived model schema, with type narrowing
+ */
+export const isCombinedSchema = (
+  schema: DataSchemaInput
+): schema is DerivedCombinedSchema => {
+  return (
+    schema !== null &&
+    typeof schema === 'object' &&
+    'schemas' in schema &&
+    Array.isArray(schema.schemas)
+  );
+};
+
+// DO NOT EDIT THE FOLLOWING VALUES, UPDATES TO DB TYPE OR STRATEGY WILL RESULT IN DB REPROVISIONING
```

Comment:
Instead of a "DO NOT EDIT..." comment, can we explain what these values are associated to with a link to the other block of code and/or docs page they need to line up with?

Reply:
Relevant category-api code: https://github.com/aws-amplify/amplify-category-api/blob/fd7f6fbc17c199331c4b04debaff69ea0424cd74/packages/amplify-graphql-api-construct/src/model-datasource-strategy-types.ts#L38-L41

Reply:
FWIW that was part of the original implementation by data-buildtime. It references this interface: https://github.com/aws-amplify/amplify-category-api/blob/main/packages/amplify-graphql-api-construct/src/model-datasource-strategy-types.ts#L38-L41. 

If there's more PR feedback I need to address, I'll modify the comment with the link. Otherwise, I'll follow up in a separate PR.